src/config/data-source.ts

```diff
@@ -0,0 +1,23 @@
+import * as dotenv from "dotenv";
+import { DataSource, DataSourceOptions } from "typeorm";
+
+dotenv.config();
+export const dataSourceOptions: DataSourceOptions = {
+  type: "mysql",
+  host: process.env.DB_HOST,
+  port: parseInt(process.env.DB_PORT, 10),
+  username: process.env.DB_USERNAME,
+  password: process.env.DB_PASSWORD,
+  database: process.env.DB_DATABASE,
+  timezone: process.env.DB_TIMEZONE,
+  entities: [__dirname + "/../**/*.entity.js"],
+  migrations: [__dirname + "/../database/migrations/*.js"],
+  extra: {
+    charset: "utf8mb4_unicode_ci",
+  },
+  synchronize: false,
+  logging: false,
+};
```

Comment:
may be able to load config asynchronously to inject config service, instead of use dotenv directly. see https://docs.nestjs.com/techniques/database#async-configuration

Reply:
I do not follow the official website, because I want to enable migration, so need to from main.js take it out.

Additional：
The settings in  typeORM v2 use TypeOrmModuleOptions, but in typeORM v3 need use DataSource.
https://github.com/typeorm/typeorm/issues/9286

Reply:
got it